Find a functionally equivalent code:android.net.nsd.NsdServiceInfo.getTxtRecord:COMMENT
Method Modifier: public      hidden      
Comment:/**
 * @hide
 */

@@ -1,3 +1,27 @@
 {
-    return mTxtRecord;
+    int txtRecordSize = getTxtRecordSize();
+    if (txtRecordSize == 0) {
+        return null;
+    }
+    byte[] txtRecord = new byte[txtRecordSize];
+    int ptr = 0;
+    for (Map.Entry<String, byte[]> entry : mTxtRecord.entrySet()) {
+        String key = entry.getKey();
+        byte[] value = entry.getValue();
+        // One byte to record the length of this key/value pair.
+        txtRecord[ptr++] = (byte) (key.length() + (value == null ? 0 : value.length) + 1);
+        // The key, in US-ASCII.
+        // Note: use the StandardCharsets const here because it doesn't raise exceptions and we
+        // already know the key is ASCII at this point.
+        System.arraycopy(key.getBytes(StandardCharsets.US_ASCII), 0, txtRecord, ptr, key.length());
+        ptr += key.length();
+        // US-ASCII '=' character.
+        txtRecord[ptr++] = (byte) '=';
+        // The value, as any raw bytes.
+        if (value != null) {
+            System.arraycopy(value, 0, txtRecord, ptr, value.length);
+            ptr += value.length;
+        }
+    }
+    return txtRecord;
 }

